### PR TITLE
Fix status of undefined

### PR DESCRIPTION
**Pull request by ka1zen**
Fix status of undefined when axios is throwing an non AxiosError

### CODE REVIEW

File: src/luminator.ts [hunk 1]
```diff
@@ -57,6 +57,17 @@ class Luminator {
     return Math.trunc(crypto.randomBytes(8).readUInt32LE(1) / 0xffffffff * 1000000);
   }
 
+  /**
+   * return status if it is a valid AxiosError
```
Comment (rimiti) [inline src/luminator.ts]: @description Returns status if it's a valid AxiosError

File: src/luminator.ts [hunk 2]
```diff
@@ -57,6 +57,17 @@ class Luminator {
...
+   * @param error
+   * @return status || -1
```
Comment (rimiti) [inline src/luminator.ts]: @return {number}

File: src/luminator.ts [hunk 3]
```diff
@@ -57,6 +57,17 @@ class Luminator {
     return Math.trunc(crypto.randomBytes(8).readUInt32LE(1) / 0xffffffff * 1000000);
   }
 
+  /**
+   * return status if it is a valid AxiosError
+   * @param error
```
Comment (rimiti) [inline src/luminator.ts]: `@param error {axios.AxiosError}`

File: src/luminator.ts [hunk 4]
```diff
@@ -57,6 +57,17 @@ class Luminator {
...
+   * @return status || -1
+   */
+  private static getStatusFromAxiosError (error: axios.AxiosError): number {
+    const hasStatus: boolean = error.hasOwnProperty('response') && error.response.hasOwnProperty('status');
+
+    return hasStatus ? error.response.status : -1;
```
Comment (rimiti) [inline src/luminator.ts]: It isn't a good to return -1. Are you sure that it's useful?

File: __tests__/unit/index.ts [hunk 5]
```diff
@@ -115,4 +115,18 @@ describe('Luminator', () => {
       expect(spy).toHaveBeenCalledTimes(2);
     });
   });
+
+  describe('Should throw all errors which is not an axios error', () => {
+    it('Should throw non axios responses error', async () => {
```
Comment (rimiti) [inline __tests__/unit/index.ts]: `'Should throw a non axios responses error'`

File: __tests__/unit/index.ts [hunk 6]
```diff
@@ -115,4 +115,18 @@ describe('Luminator', () => {
       expect(spy).toHaveBeenCalledTimes(2);
     });
   });
+
+  describe('Should throw all errors which is not an axios error', () => {
```
Comment (rimiti) [inline __tests__/unit/index.ts]: `Should throw all errors which are not an axios error`

File: src/luminator.ts (hunk 4 above)
Comment (ka1zen) [reply src/luminator.ts]: i was hesitating between 0 and -1, i've chooed the -1 to tell that there is no status (false)

**Comment by rimiti** [comment]
LGTM, well done. 🎖

**Pull request merged**